### PR TITLE
ENH add GSFitsWCS

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the GSFitsWCS classes.

### CODE REVIEW

**Comment by beckermr** [comment]
@ismael-mendoza this one is ready for review!

File: jax_galsim/utilities.py [hunk 1]
```diff
@@ -159,3 +161,88 @@ def unweighted_shape(arg):
...
+    if triangle and coefs.shape[0] != coefs.shape[1]:
+        raise GalSimIncompatibleValuesError(
+            "coefs must be square if triangle is True", coefs=coefs, triangle=triangle
+        )
+
+    def _sub(arr, i):
+        return arr[:i]
+
+    if triangle:
+        res = jax.lax.fori_loop(
```
Comment (ismael-mendoza) [inline jax_galsim/utilities.py:217]: the part below is challenging to read, would it be possible to split it up into several statements and add comments?

File: jax_galsim/utilities.py [hunk 2]
```diff
@@ -159,3 +161,88 @@ def unweighted_shape(arg):
...
+@_wraps(_galsim.utilities.horner)
+def horner(x, coef, dtype=None):
+    x = jnp.array(x)
+    coef = jnp.atleast_1d(coef)
+    res = jnp.zeros_like(x, dtype=dtype)
+
+    if len(coef.shape) != 1:
+        raise GalSimValueError("coef must be 1-dimensional", coef)
+
+    # TODO: we cannot trim zerosin jax galsim because then for all zeros
```
Comment (ismael-mendoza) [inline jax_galsim/utilities.py:175]:
```suggestion
    # TODO: we cannot trim zeros in jax galsim because then for all zeros
```

**Review by ismael-mendoza — CHANGES_REQUESTED** [review]
Almost done, just a comment in the horner2d method which is a bit challenging to review

File: tests/jax/test_api.py [hunk 3]
```diff
@@ -613,10 +685,27 @@ def test_api_wcs():
...
+        if (
+            inspect.isclass(cls)
+            and issubclass(cls, jax_galsim.wcs.BaseWCS)
+            and cls
+            not in (
+                jax_galsim.wcs.BaseWCS,
+                jax_galsim.wcs.EuclideanWCS,
+                jax_galsim.wcs.LocalWCS,
+                jax_galsim.wcs.UniformWCS,
+                jax_galsim.wcs.CelestialWCS,
```
[the remark this answers is not in the document]
Comment (beckermr) [reply tests/jax/test_api.py:704]: Yes that's correct.

File: jax_galsim/fitswcs.py [hunk 4]
```diff
@@ -0,0 +1,1075 @@
...
+                        (2.0 * m - 1.0) * a * xm[m - 1] - (m - 1.0) * xm[m - 2]
+                    ) / m
+                    xm[m, 1:] += ((2.0 * m - 1.0) * b * xm[m - 1, :-1]) / m
+                    ym[m] = (
+                        (2.0 * m - 1.0) * c * ym[m - 1] - (m - 1.0) * ym[m - 2]
+                    ) / m
+                    ym[m, 1:] += ((2.0 * m - 1.0) * d * ym[m - 1, :-1]) / m
+
+            pv2 = np.dot(xm.T, np.dot(pv, ym))
+            return pv2
```
[the remark this answers is not in the document]
Comment (beckermr) [reply jax_galsim/fitswcs.py:578]: This cast is done above at line 468.

File: jax_galsim/utilities.py (hunk 1 above)
Comment (beckermr) [reply jax_galsim/utilities.py:217]: yes happy to do so! this method is indeed quite dense!

Comment (beckermr) [reply jax_galsim/utilities.py]: done!

**Comment by beckermr** [comment]
OK @ismael-mendoza. back to you!

**Comment by beckermr** [comment]
@ismael-mendoza shall we merge?

**Review by ismael-mendoza — APPROVED** [review]
Yup! makes sense to me now

**Pull request merged**